src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentManagerActor.scala

```diff
@@ -145,11 +146,12 @@ class DeploymentManagerActor(
     case CancelDeployment(plan) =>
       sender() ! cancelDeployment(plan.id)
 
-    case DeploymentFinished(plan) =>
-      runningDeployments.remove(plan.id).map { _ =>
+    case DeploymentFinished(plan, result) =>
+      runningDeployments.remove(plan.id).map { deploymentInfo =>
         logger.info(s"Removing ${plan.id} for ${plan.targetIdsString} from list of running deployments")
         deploymentStatus -= plan.id
-        deploymentRepository.delete(plan.id)
+        deploymentRepository.delete(plan.id) // TODO: wait for future?
```

Comment:
Should it stay as a fire and forget?

Reply:
No, I've built `DeploymentManagerActor` to *not* block. Let's keep it this way pls

Reply:
I would not block but fulfill the promise the the deployment was deleted. Let's keep it as it is for now.